src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -344,26 +344,21 @@ private BoundExpression CreateTupleConversion(CSharpSyntaxNode syntax, BoundTupl
 
             NamedTypeSymbol targetType = (NamedTypeSymbol)destinationWithoutNullable;
 
-            NamedTypeSymbol targetUnderlyingType;
             if (targetType.IsTupleType)
             {
                 var destTupleType = (TupleTypeSymbol)targetType;
-                targetUnderlyingType = destTupleType.UnderlyingTupleType;
                 // do not lose the original element names in the literal if different from names in the target
+
+                // PROTOTYPE(tuples): Come back to this, what about locations?
                 targetType = destTupleType.WithElementNames(sourceTuple.ArgumentNamesOpt);
```

Comment:
This code appears to be changing the destination type. It results in a bound expression whose type isn't always the provided "destination" type. Because it doesn't follow the usual pattern of a conversion resulting in an expression of the converted-to type, I don't see how that fits into the usual type and conversion machinery.